packages/core/src/context.ts

```diff
@@ -22,6 +22,10 @@ import {
   toElicitationRequestedSchema,
 } from "./validation.js";
 
+interface Logger {
```

Comment:
we already have a logger interface defined in `packages/core/src/core.ts`, this one seems like a straggler